packages/babel-plugin-transform-parameters/src/params.js

```diff
@@ -76,11 +76,14 @@ export default function convertFunctionParams(path, loose) {
           switch (node.type) {
             case "VariableDeclarator":
               if (node.init === null) {
-                redeclarator.remove();
-              } else {
-                state.iife = true;
+                // for (var x in {}); should not be removed
+                const declaration = redeclarator.parentPath;
+                if (!declaration.parentPath.isFor({ left: declaration.node })) {
```

Comment:
In `for (...;...;...)` statements `left` doesn't exist: it's called `init`.
If you meant to only check for-of and for-in (but I don't think so), then you need `ForXStatement`.

:bulb: Idea: an ESLint plugin that prevents us from checking non-existent properties (like in this case).